pkg/reconciler/awssnssource/subscription.go

```diff
@@ -187,72 +184,6 @@ func (r *Reconciler) ensureUnsubscribed(ctx context.Context) error {
 		"Unsubscribed from SNS topic %q", topicARN)
 }
 
-// newSNSClient returns a new SNS client for the given region using static credentials.
-func newSNSClient(cli coreclientv1.SecretInterface,
-	region string, creds *v1alpha1.AWSSecurityCredentials) (*sns.SNS, error) {
-
-	credsValue, err := awsCredentials(cli, creds)
-	if err != nil {
-		return nil, fmt.Errorf("reading AWS security credentials: %w", err)
-	}
-
-	cfg := session.Must(session.NewSession(aws.NewConfig().
-		WithRegion(region).
-		WithCredentials(credentials.NewStaticCredentialsFromCreds(*credsValue)),
-	))
-
-	return sns.New(cfg), nil
-}
-
-// awsCredentials returns the AWS security credentials referenced in the
-// source's spec.
-func awsCredentials(cli coreclientv1.SecretInterface,
-	creds *v1alpha1.AWSSecurityCredentials) (*credentials.Value, error) {
-
-	accessKeyID := creds.AccessKeyID.Value
-	secretAccessKey := creds.SecretAccessKey.Value
-
```

Comment:
All this code was relocated to `client.go` almost unchanged. It's now attached to the `clientGetterWithSecretGetter` type.